Mathlib/CategoryTheory/Sites/Coverage.lean

```diff
@@ -285,6 +285,28 @@ theorem toGrothendieck_eq_sInf (K : Coverage C) : toGrothendieck _ K =
     intro X S hS
     apply saturate.of _ _ hS
 
+/-- The union of two coverages is a coverage. -/
+@[simps]
+def union (x y : Coverage C) : Coverage C where
+  covering B := x.covering B ∪ y.covering B
+  pullback := by
+    rintro X Y f S (hx | hy)
+    · obtain ⟨T, hT⟩ := x.pullback f S hx
+      exact ⟨T, Or.inl hT.1, hT.2⟩
+    · obtain ⟨T, hT⟩ := y.pullback f S hy
+      exact ⟨T, Or.inr hT.1, hT.2⟩
```

Comment:
I think it would be best to drop the definition altogether, as
```suggestion
/-- The union of two coverages is a coverage. -/
instance : Sup (Coverage C) where
  sup x y :=
  { covering B := x.covering B ∪ y.covering B
    pullback := by
      rintro X Y f S (hx | hy)
      · obtain ⟨T, hT⟩ := x.pullback f S hx
        exact ⟨T, Or.inl hT.1, hT.2⟩
      · obtain ⟨T, hT⟩ := y.pullback f S hy
        exact ⟨T, Or.inr hT.1, hT.2⟩ }
```
You'll unfortunately need to write the `simps` lemma manually, but you're currently missing the lemma spelt in terms of `\sup` anyway.

Reply:
I would like to keep the `union` notation. It feels more natural to me than `sup`. I can write the `simps` lemma in terms of `\sup` though

Reply:
The (minor) problem with having `union` at all is that whenever you use it, you can't apply any of the lemmas from `SemilatticeSup` without first using ` rw [←sup_eq_union]`.

Merging the two removes this unnecessary extra step, and is consistent with the approach used in most other places in mathlib (the exception being Set and Finset, which duplicate all the API instead)

Reply:
Ok, I'm convinced. I'll remove `union`